lib/deps/request-browser.js

```diff
@@ -3,7 +3,105 @@
 var createBlob = require('./blob.js');
 var utils = require('../utils');
 
-module.exports = function(options, callback) {
+function wrappedFetch() {
+  var args = Array.prototype.slice.call(arguments); // arguments to array
+
+  var wrappedPromise = {};
+
+  var promise = new global.Promise(function(resolve, reject) {
+    wrappedPromise.resolve = resolve;
+    wrappedPromise.reject = reject;
+  });
+
+  wrappedPromise.then = promise.then.bind(promise);
+  wrappedPromise.catch = promise.catch.bind(promise);
+  wrappedPromise.promise = promise;
+
+  global.fetch.apply(null, args).then(function(response) {
```

Comment:
Referring to `global` at all is dangerous due to Node Webkit (currently untested... I know). Better just to say `fetch`.